src/ast/passes/codegen_llvm.cpp

```diff
@@ -2917,6 +2917,11 @@ ScopedExpr CodegenLLVM::visit(Probe &probe)
                                               { b_.getPtrTy() }, // ctx
                                               false);
 
+  // Skip if we've generated too many.
+  uint64_t max_bpf_progs = bpftrace_.config_->get(ConfigKeyInt::max_bpf_progs);
+  if (probe_count_ > max_bpf_progs)
```

Comment:
Why was this added here and why don't we want the same error as below?

Reply:
The error below will have already been emitted. This was previously relying on an exception being thrown which goes all the way to main. Now the pass will complete, and actually put a diagnostic on the probe where we encountered the error.

Reply:
Maybe a comment here would be good but non-blocking.

Reply:
Added clarifying comment.